File: client/src/services/news.js
import Service from '../base/Service.js'

class NewsService extends Service {
  prefix = 'news';

  create = ({title, text}) => this._call('create', {title, text});
  remove = (id) => this._call('remove', {id});
  // get = (id) => this._call('get', {id});
  getList = () => this._call('get_list', {});
}

class NewsStorage {
  constructor() {
    this.data = [{
      title: 'Test News',
      text: 'text of test news'
    }];
  }


  async create({title, text}) {
    this.data.unshift({title, text});
    return this.data.length;
  }

  async getList() {
    return this.data;
  }
}

export default new NewsService();

File: server/skill_area/services/news.py
from skill_area.base import Service
from skill_area.models import News
from django.core.mail import send_mail


class NewsService(Service):
    def create(self, _, title, text):
        news = News(
            title=title,
            text=text
        )
        news.save()

        # send_mail(
        #     'Add news [{}]: {}'.format(news.id, news.title),
        #     news.text,
        #     'info@skillarea.ru',
        #     ['viewsharp@yandex.ru'],
        #     fail_silently=False,
        # )

        return {
            'id': news.id,
            'title': news.title,
            'text': news.text,
            'created_at': news.created_at
        }

    def get(self, session, id=None):
        news = News.objects.get(id=id)

        return {
            'id': news.id,
            'title': news.title,
            'text': news.text,
            'created_at': news.created_at
        }

    def get_list(self, _):
        newss = News.objects.all().order_by('-created_at')
        return [{
            'id': news.id,
            'title': news.title,
            'text': news.text,
            'created_at': news.created_at
        } for news in newss]

    def remove(self, id):
        news = News.objects.get(id=id)
        news.delete()
